zilliqa/src/api/types/zil.rs

```diff
@@ -67,6 +72,74 @@ struct TxBlockHeader {
     version: u32,
 }
 
+#[derive(Clone, Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct GetTxResponse {
+    #[serde(rename = "ID")]
+    id: String,
+    version: String,
+    nonce: String,
+    to_addr: H160,
+    sender_pub_key: String,
+    amount: String,
+    signature: String,
+    receipt: GetTxResponseReceipt,
+    gas_price: String,
+    gas_limit: String,
+    code: String,
+    data: String,
+}
+
+#[derive(Clone, Serialize, Debug)]
+#[serde(rename_all = "PascalCase")]
+pub struct CreateTransactionResponse {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub contract_address: Option<H160>,
+    pub info: String,
+    #[serde(rename = "TranID")]
+    pub tran_id: H256,
+}
+
+#[derive(Clone, Serialize, Debug)]
+struct GetTxResponseReceipt {
+    cumulative_gas: String,
+    epoch_num: String,
+    success: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    event_logs: Option<Vec<Value>>,
+}
+
+impl GetTxResponse {
+    pub fn new(verified_tx: VerifiedTransaction, receipt: TransactionReceipt) -> Option<Self> {
+        match verified_tx.tx {
+            SignedTransaction::Zilliqa { ref tx, .. } => Some(GetTxResponse {
+                id: verified_tx.hash.to_string(),
+                version: "65537".to_string(),
```

Comment:
Doesn't this need to consider the chain ID?

Reply:
That is a good point

Reply:
Should we add a `TODO` or raise an issue to fix this?

Reply:
I will create some tickets off the back of this PR